### PR TITLE
[Quest] convert Black Market to IF

### PR DESCRIPTION
**Pull request by jamesbradleym**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Converts Other Areas quest to IF

Brought over variable fame reward based on turn ins.

[Capture](https://www.youtube.com/watch?v=IjKPs1MaNbs) does not confirm if zoning is required or not (as coded in ASB) but I have not come across a guide suggesting it is required to repeat(in or out of era 🤷‍♂️).

## Steps to test these changes

Complete/Repeat Black Market

### CODE REVIEW

File: scripts/quests/outlands/Black_Market.lua [hunk 1]
```diff
@@ -0,0 +1,139 @@
+-----------------------------------
+-- Black Market
+-----------------------------------
+-- Log ID: 5, Quest ID: 130
+-----------------------------------
+local quest = Quest:new(xi.questLog.OUTLANDS, xi.quest.id.outlands.BLACK_MARKET)
```
Comment (claywar) [inline scripts/quests/outlands/Black_Market.lua:7]: Can you please ensure that you're commenting relevant NPCs and position values for testing this quest? (see other quest formatting)

File: scripts/quests/outlands/Black_Market.lua [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+        [xi.zone.NORG] =
+        {
+            ['Muzaffar'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:event(16)
+                end,
+
+                onTrade = function(player, npc, trade)
```
Comment (claywar) [inline scripts/quests/outlands/Black_Market.lua:54]: Minor thing of preference, but I like to sort these functions alphabetically, so onTrade first.

File: scripts/quests/outlands/Black_Market.lua [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+                        return quest:progressEvent(18, 1200, 1200)
+                    elseif npcUtil.tradeHasExactly(trade, { { xi.item.SOUTHERN_MUMMY, 4 } }) then
+                        return quest:progressEvent(19, 1201, 1201)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [17] = function(player, csid, option, npc)
```
Comment (claywar) [inline scripts/quests/outlands/Black_Market.lua:116]: While not overly necessary, there's a lot of repeated code here, which could probably be tabled

File: scripts/quests/outlands/Black_Market.lua (hunk 1 above)
Comment (jamesbradleym) [reply scripts/quests/outlands/Black_Market.lua:7]: Sure thing!

File: scripts/quests/outlands/Black_Market.lua (hunk 2 above)
Comment (jamesbradleym) [reply scripts/quests/outlands/Black_Market.lua:54]: Done.

File: scripts/quests/outlands/Black_Market.lua (hunk 3 above)
Comment (jamesbradleym) [reply scripts/quests/outlands/Black_Market.lua:116]: Can do! - let me know if I should nix the quest reward title and just throw it into the new handleTurnIn method.

**Comment by jamesbradleym** [comment]
Closing this PR to reduce open PRs and prevent staleness—will revisit.

**Pull request closed without merging**